src/drivers/osd/atxxxx/atxxxx.cpp

```diff
@@ -455,7 +401,7 @@ OSDatxxxx::update_screen()
 		ret |= add_flighttime(flight_time_sec, 1, 14);
 
 	} else {
-		flight_mode = get_flight_mode(_nav_state);
+		flight_mode = mode_util::nav_state_names[_nav_state];
```

Comment:
I really don't think we should have modules calling into each other directly.

I would either move this little piece of mode_util to a library (src/lib) or at this point just include the mode string in vehicle_status (it'll soon need to be dynamic like this anyway).

Reply:
I'm moving it to libs in https://github.com/PX4/PX4-Autopilot/pull/20707/commits/e1728c547302f67df4b4fabb28395e94d39147c7.
We can also add it to a topic.

Reply:
> we should have modules calling into each other directly

No we shouldn't. I saw it more as a helper library but you're right, it should then be one.

> I'm moving it to libs in https://github.com/PX4/PX4-Autopilot/commit/e1728c547302f67df4b4fabb28395e94d39147c7.

Nice! I don't want to interfere with that 👍 
I'll close this and if at all suggest on top of your changes 🙏